File: Domovoy/Models/ApartmentHouse.cs
namespace Domovoy.Models;

public class ApartmentHouse
{
    public int Id { get; set; }
    
    public Address Address { get; set; }
    
    public List<HouseEntrance> HouseEntrances { get; set; }
    public ResidentialComplex ResidentialComplex { get; set; }
    public int ResidentialComplexId { get; set; }

    public int ChatId { get; set; }
    public Chat Chat { get; set; }

    public List<ApartmentService> ApartmentServices { get; set; }
    public List<InformerService> InformerServices { get; set; }

    public override string ToString() => $"Дом {Address}";
}

public class ApartmentHouseViewModel
{
    public int Id { get; set; }
    
    public Address Address { get; set; }
    
    public ResidentialComplexViewModel ResidentialComplex { get; set; }
}

public class ApartmentHouseDetails
{
    public int Id { get; set; }
    
    public Address Address { get; set; }
}

public class ApartmentHouseCreate
{
    public Address Address { get; set; }
    
    public int ResidentialComplexId { get; set; }
}

File: Domovoy/ClientApp/src/api/models/ApartmentHouse.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Address } from './Address';
import type { ApartmentService } from './ApartmentService';
import type { HouseEntrance } from './HouseEntrance';
import type { InformerService } from './InformerService';
import type { ResidentialComplex } from './ResidentialComplex';

export type ApartmentHouse = {
    id?: number;
    address?: Address;
    houseEntrances?: Array<HouseEntrance> | null;
    residentialComplex?: ResidentialComplex;
    residentialComplexId?: number;
    apartmentServices?: Array<ApartmentService> | null;
    informerServices?: Array<InformerService> | null;
};